From 27040b1175b670e8b8ae1d50c295cc3c4642bacb Mon Sep 17 00:00:00 2001
From: Wojciech Bartczak <wbartczak@marvell.com>
Date: Thu, 10 Dec 2020 19:06:11 -0800
Subject: [PATCH 1221/1921] driver: mmc: Adds new overrides for eMMC bus output
 timings

The change provieds new configuration options for following eMMC bus modes:
- Legacy
- MMC HS (SDR)
- MMC DDR52

The goal is to allow better testing of all supported eMMC modes.
Additionally the change allows users to customize output timings
for specific board without code change.

Signed-off-by: Wojciech Bartczak <wbartczak@marvell.com>
Change-Id: Ic96fab1467d74a3970800ec884eab302e4594ca7
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/44770
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/mmc/host/cavium.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/mmc/host/cavium.c b/drivers/mmc/host/cavium.c
index 560f8cd1c0ac..89c601abc611 100644
--- a/drivers/mmc/host/cavium.c
+++ b/drivers/mmc/host/cavium.c
@@ -2171,6 +2171,19 @@ static int cvm_mmc_of_parse(struct device *dev, struct cvm_mmc_slot *slot)
 			     &slot->cmd_out_taps_dly[MMC_TIMING_MMC_HS400]);
 	of_property_read_u32(node, "marvell,data-out-hs400-dly",
 			     &slot->data_out_taps_dly[MMC_TIMING_MMC_HS400]);
+	of_property_read_u32(node, "marvell,cmd-out-hs-sdr-dly",
+			     &slot->cmd_out_taps_dly[MMC_TIMING_MMC_HS]);
+	of_property_read_u32(node, "marvell,data-out-hs-sdr-dly",
+			     &slot->data_out_taps_dly[MMC_TIMING_MMC_HS]);
+	of_property_read_u32(node, "marvell,cmd-out-hs-ddr-dly",
+			     &slot->cmd_out_taps_dly[MMC_TIMING_MMC_DDR52]);
+	of_property_read_u32(node, "marvell,data-out-hs-ddr-dly",
+			     &slot->data_out_taps_dly[MMC_TIMING_MMC_DDR52]);
+	of_property_read_u32(node, "marvell,cmd-out-legacy-dly",
+			     &slot->cmd_out_taps_dly[MMC_TIMING_LEGACY]);
+	of_property_read_u32(node, "marvell,data-out-legacy-dly",
+			     &slot->data_out_taps_dly[MMC_TIMING_LEGACY]);
+
 	max_frequency = max_supported_frequency(slot->host);
 
 	/* Set maximum and minimum frequency */
-- 
2.31.1

